tests/driver_ph_oem/main.c

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2019 University of Applied Sciences Emden / Leer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the Atlas Scientific pH OEM sensor driver
+ *
+ * @author      Igor Knippenberg <igor.knippenberg@gmail.com>
+ *
+ * @}
+ */
+
+#include "xtimer.h"
+#include "timex.h"
+
+#include "ph_oem.h"
+#include "ph_oem_params.h"
+#include "ph_oem_regs.h"
+
+#define SLEEP_SEC                   (5)
+
+/* off by default, so it won't reset your previous calibration */
+#define CALIBRATION_TEST_ENABLED    (false)
+
+static ph_oem_t dev;
+
+static void interrupt_pin_cb(void *arg)
+{
+    puts("\n[IRQ - Reading done]");
+
+    /* stop pH sensor from taking further readings*/
+    ph_oem_set_device_state(&dev, PH_OEM_STOP_READINGS);
+
+    /* reset interrupt pin in case of falling or rising flank */
+    if (dev.params.irq_option != PH_OEM_IRQ_BOTH) {
+        ph_oem_reset_interrupt_pin(&dev);
```

Comment:
Is it not valid to call this function when `PH_OEM_IRQ_BOTH` is being used? If so maybe check for the case in the function and return directly.

Reply:
Sorry, just realized that this callback is being called in [interrupt context](http://doc.riot-os.org/group__drivers__periph__gpio.html#ga48ba4c318486db320a4bf6237b49d61c), and that none of this functions (which access the I2C bus) should be called here. Instead use some mechanism such as [events](http://doc.riot-os.org/group__sys__event.html)

Reply:
You are right, haven't thought about that. I added an event queue which gets claimed from a dedicated thread that loops the event queue. An event will be posted in the interrupt callback.

Reply:
I think for the test application I would go with a simpler solution, just calling `event_wait` from the main thread if the interrupt pin is being used. Even just `thread_flags_set` and `thread_flags_wait_one` could be used. No need for a second thread running.

Reply:
Removed the thread. Waiting with `event_wait` for the event to be posted by the interrupt callback.